arm/examples/README.md

```diff
@@ -0,0 +1,67 @@
+# ARM Examples in Go
+
+## Configuring your environment
+
+Install the Azure CLI tool and login using the following commands.
+
+```bash
+$ curl -L https://aka.ms/InstallAzureCli | bash
+$ exec -l $SHELL
+$ az login
+```
+
+Create a `Service Principal` for the Azure Active Directory using the following command.
```

Comment:
In general, I've been trying to think about how we could reduce the Authentication burden of using out samples. Optionally, we could skip creating a Service Principal and instead just use the token that is created by the CLI above. We already have a few customers following that pattern, and there is code to support it that can be found here:
[github.com/Azure/go-autorest/autorest/azure/cli](https://github.com/Azure/go-autorest/blob/c0eb859387e57a164bf64171da307e2ef8168b58/autorest/azure/cli/token.go). Yet another option would be to use device-auth from the sample itself: [github.com/Azure/go-autorest/autorest/adal#InitiateDeviceAuth](https://github.com/Azure/go-autorest/blob/c16bc27650917ee2e5e6b7c87e68ff3303793303/autorest/adal/devicetoken.go#L104) One downside is that we would need to go query Azure for the available Tenants/Subscriptions as part of running the sample, or continue to read from the Tenant/Subscription environment variables.

However, I'm willing to believe that there is some intrinsic value in showing folks how to use Service Principal. I'd hear you out if you have strong opinions there.